qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -0,0 +1,111 @@
+        """Set z2symmetry_reduction"""
+        self._z2symmetry_reduction = z2symmetry_reduction
+        self._z2symmetries = None
+
+    @property
+    def z2symmetries(self) -> Optional[Z2Symmetries]:
+        """Get z2symmetries. Will be `None` until :meth:`to_qubit_ops` has been called."""
+        return self._z2symmetries
+
+    def to_qubit_ops(self, second_q_ops: List[SecondQuantizedOp],
+                     z2symmetry_reduction: List[int] = None,
+                     ) -> List[Optional[PauliSumOp]]:
+        """
+        Maps the given list of second quantized operators to qubit operators. Also it will
+        carry out z2 symmetry reduction on the qubit operators if z2symmetry_reduction has
+        been specified.
+
+        Args:
+            second_q_ops: The list of second quantized operators to be converted
+            z2symmetry_reduction: Optional z2symmetry reduction, the sector of the symmetry
+
+        Returns:
+            A list of qubit operators or the same length as the second_q_ops list. The first
+            operator in the second_q_ops list is treated as the main operator and others must
+            commute with its symmetry, when symmetry reduction is being done. If it does not
+            then the position in the output list will be set to `None` to preserve the order.
+        """
+        if z2symmetry_reduction is not None:
+            self.z2symmetry_reduction = z2symmetry_reduction
+
+        qubit_ops = self._map_to_qubits(second_q_ops)
+        qubit_ops_reduced = self._symmetry_reduce(qubit_ops)
+
+        return qubit_ops_reduced
+
+    def _map_to_qubits(self, second_q_ops: List[SecondQuantizedOp]) -> List[PauliSumOp]:
+        if isinstance(self._mappers, tuple):
+            raise NotImplementedError
+
+        qubit_ops: List[PauliSumOp] = []
+        main_op = second_q_ops[0]
+        if main_op._fermion is not None:
+            for op in second_q_ops:
+                if op._fermion is None:
+                    raise QiskitNatureError("Second quantized operators must contain same type")
+                qubit_ops.append(self._mappers.map(op._fermion))
+        else:
+            raise QiskitNatureError("Cannot map invalid second quantized operator")
+
+        return qubit_ops
+
+    def _symmetry_reduce(self, qubit_ops: List[PauliSumOp]) -> List[Optional[PauliSumOp]]:
+
+        if self.z2symmetry_reduction is None:
+            reduced_qubit_ops = qubit_ops
+            self._z2symmetries = Z2Symmetries([], [], [], None)
+        else:
+            # If we are given the symmetry sector we can use it to reduce all operators.
+            # Otherwise if we are given a callable that can determine the right sector we
+            # will use that. Note this callable will most likely need problem related data
+            # such as info contained within MolecularProblem, so we need access in some way
+            # which needs to be determined.
```

Comment:
Yes, problem-specific information needs to be passed into here. Thus, this needs to happen at runtime. We will discuss later offline and I will update here afterwards :+1:

Reply:
The thing about the callable was an initial thought. I dropped that idea for just being provided the sector either when the converter is created or when the 2nd quant ops are converted to qubit_ops are created (the latter would override anything that had been passed in. This way an end user can manually configure the convertor for the sector as well as the stack (algorithms) using it can auto determine a sector and ask for that to be used. Since I was not focussing on Z2Symmetries I had thought about it and made these comments but forgot to delete them since I did not go back to that yet.